cdap-common/bin/cdap

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+#
+# Copyright © 2016 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+#
+# This script is the main script for performing all CDAP functions. With it,
+# you can start, stop, or get status on CDAP and its services, run queries,
+# execute CLI commands, and more. Anything not handled by this script is passed
+# to the CDAP CLI to be interpreted. 
+#
+
+__script=${BASH_SOURCE[0]} __script_bin=$(dirname ${__script})
+__comp_home=${__script%/*/*}
+if [[ ${__comp_home%/*} == /opt/cdap ]]; then
+  __app_home=${__comp_home}
+  __cdap_home=/opt/cdap
+else
+  __app_home=$(dirname ${__script_bin})
+  __cdap_home=${__app_home}
+fi
+
+export CDAP_HOME=${__cdap_home}
+
+# Source our functions script, which will be located with us
+source "${__app_home}"/bin/functions.sh
+
+__fn_home=$(cdap_home)
+# Do a small sanity check
+if [[ ${__cdap_home} != ${__fn_home} ]]; then
+  die "CDAP_HOME doesn't match output of cdap_home function! Aborting!"
+fi
+
+# Load user-modifiable configuration
+test -f "${CDAP_CONF}"/cdap-env.sh && source "${CDAP_CONF}"/cdap-env.sh
+
+# Main script, handles options and Does the Right Thing™
+case ${1} in
+  auth-server|kafka-server|master|router|ui) CDAP_SERVICE=${1}; shift; cdap_service ${CDAP_SERVICE} ${@}; __ret=${?} ;;
+  classpath) cdap_service master classpath; __ret=${?} ;;
+  cli|shell) shift; cdap_cli ${@}; __ret=${?} ;;
+  config-tool) shift; cdap_config_tool ${@}; __ret=${?} ;;
+  run) shift; cdap_run_class ${@}; __ret=${?} ;;
+  sdk|standalone) shift; cdap_sdk ${@}; __ret=${?} ;;
+  tx-debugger) shift; cdap_tx_debugger ${@}; __ret=${?} ;;
+  -h|--help|usage|'')
+    echo
+    echo "Usage: ${0} <command> [arguments]"
+    echo
+    echo "  Commands:"
+    echo
+    echo "    auth-server       - Sends the arguments to the appropriate service"
+    echo "    kafka-server"
+    echo "    master"
+    echo "    router"
+    echo "    ui"
+    echo
+    echo "    cli               - Starts an interactive CDAP CLI session"
+    echo "    shell"
+    echo
+    echo "    config-tool       - Returns JSON encoded CDAP configuration key/value pairs"
+    echo "    run               - Runs a given class with arguments using CDAP CLASSPATH and environment"
+    echo "    tx-debugger       - Runs CDAP transaction debugger"
+    echo
+    echo "    sdk               - Sends the arguments to the SDK service"
+    echo "    standalone"
+    echo
```

Comment:
I could make the usage into a function, have that function try to detect the presence of the `${CDAP_HOME}/lib/co.cask.cdap.cdap-standalone-@@package.version@@.jar` and output the correct usage based on that file's existence. If it's alright, I would like to hold off on that until another PR, as I am working on some other changes (CDAP-7174) that use this branch as a base and I want to reduce conflicts, since that branch has to modify some of the code in this one.

I filed [CDAP-7218](https://issues.cask.co/browse/CDAP-7218) to track the changes to the help. Does that work?